src/macports1.0/reclaim.tcl

```diff
@@ -126,12 +127,42 @@ namespace eval reclaim {
                     ui_info [msgcat::mc "Skipping deletion of %s (dry run)" $root_build]
                 } else {
                     ui_info [msgcat::mc "Deleting %s" $root_build]
-                    file delete -force $root_build
+                    set builddirs [glob -nocomplain -types d [file join $root_build *]]
+                    catch {file delete -force -- {*}$builddirs}
```

Comment:
Is the catch just in case there are no build dirs and the glob thus matches nothing? Would it be worth reporting if the deletion fails for other reasons?

Reply:
yes, it will fail if there is nothing there; are there other reasons it might fail?

Reply:
I see portclean checks for errors; it would have to ignore if there are no dirs there.

Reply:
I copy/pasted the code that is in a lot of other places - I couldn't figure how to trigger it to test.

Reply:
Come to think of it, you could do the glob earlier and, if it matches nothing, report that there are no build dirs to delete and return immediately. That would avoid having to answer the interactive question in that case.

Error handling is still good to have. Look at the ERRORS section in `man 2 unlink` for a (probably not exhaustive) list of things that can go wrong.

Reply:
Ok bypassing the question sounds good if there's nothing to delete.